File: src/main/java/me/kuku/yuq/dao/QQDao.java
package me.kuku.yuq.dao;

import com.icecreamqaq.yudb.YuDao;
import com.icecreamqaq.yudb.jpa.annotation.Dao;
import com.icecreamqaq.yudb.jpa.annotation.Execute;
import com.icecreamqaq.yudb.jpa.annotation.Select;
import me.kuku.yuq.entity.QQEntity;

import java.util.List;

@Dao
public interface QQDao extends YuDao<QQEntity, Integer> {
    @Select("from QQEntity where qq = ?0 and group_id = ?1")
    QQEntity findByQQAndGroup(Long qq, Integer group);
    @Execute("delete from QQEntity where qq = ?0 and group_id = ?1")
    int delByQQAndGroup(Long qq, Integer group);
    @Select("from QQEntity")
    List<QQEntity> findAll();
}

File: src/main/kotlin/me/kuku/yuq/dao/QQDao.kt
package me.kuku.yuq.dao

import com.icecreamqaq.yudb.YuDao
import com.icecreamqaq.yudb.jpa.annotation.Dao
import com.icecreamqaq.yudb.jpa.annotation.Execute
import com.icecreamqaq.yudb.jpa.annotation.Select
import me.kuku.yuq.entity.QQEntity

@Dao
interface QQDao: YuDao<QQEntity, Int>{
    @Select("from QQEntity where qq = ?1 and group_id = ?2")
    fun findByQQAndGroup(qq: Long, group: Int): QQEntity?
    @Execute("delete from QQEntity where qq = ?1 and group_id = ?2")
    fun delByQQAndGroup(qq: Long, group: Int): Int
    @Select("from QQEntity")
    fun findAll(): List<QQEntity>
}